packages/react/src/components/Provider/Provider.tsx

```diff
@@ -3,33 +3,40 @@ import { render } from 'fela-dom'
 import * as _ from 'lodash'
 import * as PropTypes from 'prop-types'
 import * as React from 'react'
-import { RendererProvider, ThemeProvider } from 'react-fela'
+// @ts-ignore
+import { RendererProvider, ThemeProvider, ThemeContext } from 'react-fela'
 import * as customPropTypes from '@stardust-ui/react-proptypes'
 
 import {
   felaRenderer as felaLtrRenderer,
   felaRtlRenderer,
   isBrowser,
-  mergeThemes,
   ChildrenComponentProps,
 } from '../../lib'
 
 import {
   ThemePrepared,
-  ThemeInput,
   StaticStyleObject,
   StaticStyle,
   StaticStyleFunction,
   FontFace,
   ComponentVariablesInput,
+  ProviderContextInput,
+  ProviderContextPrepared,
+  Renderer,
+  ThemeInput,
 } from '../../themes/types'
 
 import ProviderConsumer from './ProviderConsumer'
 import { mergeSiteVariables } from '../../lib/mergeThemes'
 import ProviderBox from './ProviderBox'
 import { WithAsProp } from '../../types'
+import mergeContexts from '../../lib/mergeContexts'
```

Comment:
may we call the file as `mergeProviderContexts`, so it will be clear what this is about by just looking at its name? Otherwise it may be not obvious from the first look. Something like:

```js
// not necessary to change name of the imported function, 
// as `Provider`'s file context (pan not intended:) is enough here
import mergeContexts from '../../lib/mergeProviderContexts'
```